OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -764,10 +770,28 @@ void QueueCaptureOrders()
 				.OrderByDescending(target => target.Actor.GetSellValue())
 				.Take(maximumCaptureTargetOptions);
 
-			if (Info.CapturableActorTypes.Any())
+			if (Info.CapturableActorTypes.Any() && !Info.AvoidCapturingActorTypes.Any())
+			{
+				capturableTargetOptions = capturableTargetOptions.Where(target =>
+					Info.CapturableActorTypes.Contains(target.Actor.Info.Name.ToLowerInvariant()));
+				externalCapturableTargetOptions = externalCapturableTargetOptions.Where(target =>
+					Info.CapturableActorTypes.Contains(target.Actor.Info.Name.ToLowerInvariant()));
+			}
+			else if (Info.CapturableActorTypes.Any() && Info.AvoidCapturingActorTypes.Any())
+			{
+				capturableTargetOptions = capturableTargetOptions.Where(target =>
+					(Info.CapturableActorTypes.Contains(target.Actor.Info.Name.ToLowerInvariant()) &&
+					!Info.AvoidCapturingActorTypes.Contains(target.Actor.Info.Name.ToLowerInvariant())));
+				externalCapturableTargetOptions = externalCapturableTargetOptions.Where(target =>
+					(Info.CapturableActorTypes.Contains(target.Actor.Info.Name.ToLowerInvariant()) &&
+					!Info.AvoidCapturingActorTypes.Contains(target.Actor.Info.Name.ToLowerInvariant())));
+			}
+			else if (!Info.CapturableActorTypes.Any() && Info.AvoidCapturingActorTypes.Any())
 			{
-				capturableTargetOptions = capturableTargetOptions.Where(target => Info.CapturableActorTypes.Contains(target.Actor.Info.Name.ToLowerInvariant()));
-				externalCapturableTargetOptions = externalCapturableTargetOptions.Where(target => Info.CapturableActorTypes.Contains(target.Actor.Info.Name.ToLowerInvariant()));
+				capturableTargetOptions = capturableTargetOptions.Where(target =>
```

Comment:
Why do we populate `capturableTargetOptions`  when there aren't any actors that can/should be captured are defined?